Add imap custom event data max message size option (#27425)
home-assistant/home-assistant.github.io: source/_integrations/imap.markdown

folder:
  description: The IMAP folder configuration
text:

>   description: The email body `text` of the the message (by default only the first 2048 bytes will be available)

Reviewer: ```suggestion
  description: The email body `text` of the message (by default, only the first 2048 bytes will be available)
```

The `event_type` for the custom event should be set to `imap_content`. The configuration below shows how you can use the event data in a template `sensor`.

> If the default maximum message size (2048) to be used in events is too small for your needs, then this maximum size setting can be increased(advanced mode is required).

Reviewer: ```suggestion
If the default maximum message size (2048 bytes) to be used in events is too small for your needs, then this maximum size setting can be increased. You need to have your profile set to _advanced_ mode to do this.
```
advanced mode: You mean the advanced mode in the user profile? If not, please ignore my suggestion.

<div class='note warning'>

> Increasing the default maximum message size (2048) could have a negative impact on performance as event data is also logged by the `recorder`. If the total event data size is exceeds the maximum event size (32168) the event will be skipped.

Reviewer: ```suggestion
Increasing the default maximum message size (2048 bytes) could have a negative impact on performance as event data is also logged by the `recorder`. If the total event data size exceeds the maximum event size (32168 bytes), the event will be skipped.
```
I added bytes everywhere to be more explicit about the unit. But maybe there is a reason you did not add this?
